### PR TITLE
update: Change default to server-side generated revision names (extended)

### LINKED ISSUE #1144: Use server-generated revisions names as default for services (instead of BYO revision name)
As discussed in https://github.com/knative/serving/issues/9544 BYO revision names that we use by default (i.e. the client set the revision name and not the serving controller), has some issues when it comes to sink bindings, but also for `kn service apply` we can't use BYO revision names. Tbh, I always have a hard time to remember the actual use case for what BYO revision names are used (and again, I forgot about it ;-(. I conclude, that the use case for BYO revision names is so rare that it does not justified to use this as a default (when contrasting this to the trouble that it brings elsewhere).

My suggestion is to make server-side naming the default and *only* when `--revision-name` is specified we switch to BYO revision names. And the moment, switching off BYO revision naming is even very obfuscated as one has to provide a `--revision-name=` (i.e. an empty name) so switch this off.

The switch should work transparently except for that one use that I don't remember anymore (@evankanderson I think you explained it to me earlier, but I forgot again. sorry :( . From a UX perspective we just would not allow anymore a `--revision-name` without or empty argument, and just use server-side revision naming by default when this option is not given.

Opinions ?

### PR DESCRIPTION
**Pull request by rhuss**
Fixes #1144.

This is an extension of #1236 with better handling for service updates. If tests are passing then this PR should be merged instead of #1236

### CODE REVIEW

**Comment by rhuss** [comment]
@maximilien could I get this stamped please ? This is the PR that we talked about at the WG call, but a version that should be more robust (and irons out also my corner case). It checks now the generation's first before going into a wait loop. This should be much more robust as we did before anyway.

I would need this PR to be merged to continue with the 0.21 release (i will release the plugins then tomorrow, but would like to kick of the auto-release for kn already)

**Comment by rhuss** [comment]
@navidshaikh could you please approve + lgtm this PR ? When merged, I wold then kickoff the kn release.

File: CHANGELOG.adoc [hunk 1]
```diff
@@ -37,6 +37,10 @@
 | 🐣
 | Do not print `serviceUID` and `configUID` labels in service export results
 | https://github.com/knative/client/pull/1194[#1194]
+
+| ✨
+| Use server-side generated revision names by default now. For BYO revision names use `--revision name` for service commands
+| https://github.com/knative/client/issues/1144[#1144]
```
Comment (navidshaikh) [inline CHANGELOG.adoc:43]:
```suggestion
| ✨
| Use server-side generated revision names by default now. For BYO revision names use `--revision-name` for service commands
| https://github.com/knative/client/issues/1240[#1240]
```

**Review by navidshaikh — APPROVED** [review]
/lgtm
/approve

The changelog fix can be done in a follow up.

**Comment by rhuss** [comment]
Ah, good point. Fixed (and need another stamp now ;-)

**Comment by rhuss** [comment]
@navidshaikh Also stupid clabot needs some love again :)

thanks !

**Comment by navidshaikh** [comment]
@googlebot I fixed it.

**Comment by navidshaikh** [comment]
@googlebot I fixed it.

**Comment by rhuss** [comment]
> @googlebot I fixed it.

Maybe `@googlebot I consent` works better ?

If not, I'm going squash everything and push again

**Comment by navidshaikh** [comment]
@googlebot I consent.

**Comment by navidshaikh** [comment]
@rhuss : lets squash and push again

**Comment by rhuss** [comment]
@navidshaikh done. Hopefully we are at the last stamp ;-)

**Pull request merged**